unittests/api_tests.cpp

```diff
@@ -935,6 +946,54 @@ BOOST_AUTO_TEST_CASE(checktime_fail_tests) { try {
    BOOST_REQUIRE_EQUAL( t.validate(), true );
 } FC_LOG_AND_RETHROW() }
 
+BOOST_AUTO_TEST_CASE(checktime_pause_test) { try {
+
+   fc::temp_directory tempdir;
+   auto conf_genesis = tester::default_config( tempdir );
+
+   auto& cfg = conf_genesis.second.initial_configuration;
+
+   cfg.max_block_cpu_usage        = 250'000;
+   cfg.max_transaction_cpu_usage  = 4'999; // needs to be large enough for create_account and set_code
+   cfg.min_transaction_cpu_usage  = 1;
+
+   tester t( conf_genesis.first, conf_genesis.second );
+   t.execute_setup_policy( setup_policy::full );
+
+   t.produce_blocks(2);
+
+   ilog( "create account" );
+   t.create_account( "pause"_n );
+   ilog( "set code" );
+   t.set_code( "pause"_n, contracts::test_api_wasm() ); // only test that should use pause account so it is not cached
+   ilog( "produce block" );
+   t.produce_blocks(1);
+
+   int64_t ram_bytes; int64_t net; int64_t cpu;
+   auto& rl = t.control->get_resource_limits_manager();
+   rl.get_account_limits( "pause"_n, ram_bytes, net, cpu );
+   BOOST_CHECK_EQUAL( cpu, -1 );
+   auto cpu_limit = rl.get_block_cpu_limit();
+   idump(("cpu_limit")(cpu_limit));
+   BOOST_CHECK( cpu_limit <= 250'000 );
+
+   // 6ms should be lower than time to load wasm
+   auto before = fc::time_point::now();
+   BOOST_CHECK_EXCEPTION( call_test( t, test_pause_action<TEST_METHOD("test_checktime", "checktime_failure")>{},
+                                     0, 6, fc::raw::pack(10000000000000000000ULL), "pause"_n ),
+                          tx_cpu_usage_exceeded, is_tx_cpu_usage_exceeded );
+   auto after = fc::time_point::now();
+   // Test that it runs longer than specified limit of 6 to allow for wasm load time.
+   auto dur = (after - before).count();
+   dlog("elapsed ${e}us", ("e", dur) );
+   BOOST_CHECK( dur > 6500 );
```

Comment:
Why 6500?  The 6ms passed to call_test is the explicit deadline, not the billing timer.  It should not be extended by pause.  The billing timer is 5ms.

Reply:
WASM load times (on my machine) were taking about 10ms. `6500` is smaller than 10000 but higher than `4999`.